tsl/src/continuous_aggs/create.c

```diff
@@ -722,6 +726,36 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 						 errmsg(
 							 "time bucket function must reference a hypertable dimension column")));
 
+			if (list_length(fe->args) == 3)
+			{
+				/*
+				 * Currently the third argument of the bucketing function can be
+				 * only a timezone. Only immutable expressions can be specified.
+				 */
```

Comment:
Isn't it possible that the third argument is not a timezone? E.g.

```

CREATE OR REPLACE FUNCTION timescaledb_experimental.time_bucket_ng(bucket_width INTERVAL, ts DATE, origin DATE) RETURNS DATE
    AS '@MODULE_PATHNAME@', 'ts_time_bucket_ng_date' LANGUAGE C IMMUTABLE PARALLEL SAFE STRICT;

CREATE OR REPLACE FUNCTION timescaledb_experimental.time_bucket_ng(bucket_width INTERVAL, ts TIMESTAMP, origin TIMESTAMP) RETURNS TIMESTAMP
    AS '@MODULE_PATHNAME@', 'ts_time_bucket_ng_timestamp' LANGUAGE C IMMUTABLE PARALLEL SAFE STRICT;

```

Reply:
Currently - no. Origin will be supported in the next PR.